plugins/ksp/src/org/jetbrains/kotlin/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -189,6 +199,16 @@ class ResolverImpl(
         return KSNameImpl.getCached(name)
     }
 
+    override fun mapToJvmSignature(declaration: KSDeclaration): String {
+        return when (declaration) {
+            is KSFunctionDeclaration -> resolveFunctionDeclaration(declaration)?.let { typeMapper.mapAsmMethod(it).descriptor } ?: ""
+            is KSPropertyDeclaration -> resolvePropertyDeclaration(declaration)?.let {
+                typeMapper.mapFieldSignature(it.type, it) ?: typeMapper.mapType(it).descriptor
+            } ?: ""
+            else -> ""
+        }
```

Comment:
Are we talking about signatures or descriptors here?

> A descriptor is a string representing the type of a field or method. ... 

> A signature is a string representing the generic type of a field or method, or generic type information for a class declaration.


See https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.3